io.c

```diff
@@ -14700,47 +14700,286 @@ set_LAST_READ_LINE(VALUE val, ID _x, VALUE *_y)
  *
- * You can now use ARGF to work with a concatenation of each of these named
- * files. For instance, ARGF.read will return the contents of _file1_
- * followed by the contents of _file2_.
+ *     Initially, it contains the arguments and options
+ *     that are passed to it from the command line;
+ *     the running Ruby program can modify that array as it likes.
  *
- * After a file in +ARGV+ has been read ARGF removes it from the Array.
- * Thus, after all files have been read +ARGV+ will be empty.
+ * - **ARGF** may be thought of as the <b>argument files</b> object.
  *
- * You can manipulate +ARGV+ yourself to control what ARGF operates on. If
- * you remove a file from +ARGV+, it is ignored by ARGF; if you add files to
- * +ARGV+, they are treated as if they were named on the command line. For
- * example:
+ *     It can access file streams and/or the `$stdin` stream,
+ *     based on what it finds in `ARGV`.
+ *     This provides a convenient way for the command line
+ *     to specify streams for a Ruby program to read.
  *
- *     ARGV.replace ["file1"]
- *     ARGF.readlines # Returns the contents of file1 as an Array
- *     ARGV           #=> []
- *     ARGV.replace ["file2", "file3"]
- *     ARGF.read      # Returns the contents of file2 and file3
+ * ## Reading
  *
- * If +ARGV+ is empty, ARGF acts as if it contained <tt>"-"</tt> that
- * makes ARGF read from STDIN, i.e. the data piped or typed to your
- * script. For example:
+ * \ARGF may read from _source_ streams,
+ * which at any particular time are determined by the content of `ARGV`.
  *
- *     $ echo "glark" | ruby -e 'p ARGF.read'
- *     "glark\n"
+ * ### Simplest Case
+ *
+ * When the <i>very first</i> \ARGF read occurs with an empty `ARGV` (`[]`),
+ * the source is `$stdin`:
+ *
+ * - \File `t.rb`:
+ *
+ *     ```
+ *     p ['ARGV', ARGV]
+ *     p ['ARGF.read', ARGF.read]
+ *     ```
+ *
+ * - Commands and outputs
+ *   (see below for the content of files `foo.txt` and `bar.txt`):
+ *
+ *     ```sh
+ *     echo "Open the pod bay doors, Hal." | ruby t.rb
+ *     ["ARGV", []]
+ *     ["ARGF.read", "Open the pod bay doors, Hal.\n"]
+ *     ```
+ *
+ *     ```sh
+ *     cat foo.txt bar.txt | ruby t.rb
+ *     ["ARGV", []]
+ *     ["ARGF.read", "Foo 0\nFoo 1\nBar 0\nBar 1\nBar 2\nBar 3\n"]
+ *     ```
+ *
+ * ### About the Examples
+ *
+ * Many examples here assume the existence of files `foo.txt` and `bar.txt`:
+ *
+ * ```sh
+ * cat foo.txt
+ * Foo 0
+ * Foo 1
+ * cat bar.txt
+ * Bar 0
+ * Bar 1
+ * Bar 2
+ * Bar 3
+ * ```
+ *
+ * ### Sources in `ARGV`
+ *
+ * For any \ARGF read _except_ the {simplest case}[rdoc-ref:ARGF@Simplest+Case]
+ * (that is, _except_ for the <i>very first</i> \ARGF read with an empty `ARGV`),
+ * the sources are found in `ARGV`.
+ *
+ * \ARGF assumes that each element in array `ARGV` is a potential source,
+ * and is one of:
+ *
+ * - The string path to a file that may be opened as a stream.
+ * - The character `'-'`, meaning stream`$stdin`.
```

Comment:
```suggestion
 * - The character `'-'`, meaning stream `$stdin`.
```

Reply:
RDoc is not markdown.

Reply:
The document begins with this:
```
 * Document-class:  ARGF
 *
 * :markup: markdown
 *
 * ## \ARGF and `ARGV`
```
So it is markdown.  Do you mean that it should not be markdown?

Reply:
Ok, I missed it.